File: CapturarFoto/Componente/vidgrab_8.5.3.1/Delphi/DEMOS/ServiceApplication/CaptureService_.pas
unit CaptureService_;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, SvcMgr, Dialogs, VidGrab;

type
  TService1 = class(TService)
	 procedure ServiceStart(Sender: TService; var Started: Boolean);
	 procedure ServiceStop(Sender: TService; var Stopped: Boolean);
	 procedure ServiceExecute(Sender: TService);
    procedure ServiceCreate(Sender: TObject);
    procedure ServiceDestroy(Sender: TObject);
  private
	 { Private declarations }
	  VideoGrabber: TVideoGrabber;
	  procedure ExecuteService;
  public
	 function GetServiceController: TServiceController; override;
	 { Public declarations }
  end;

var
  Service1: TService1;

implementation

{$R *.DFM}

// THIS DEMO START THE PREVIEW OF THE DEFAULT VIDEO CAPTURE DEVICE AND CAPTURE FRAMES PERIODICALLY EVERY 5 SECONDS TO THE c:\mycapture FOLDER

procedure ServiceController(CtrlCode: DWord); stdcall;
begin
   Service1.Controller(CtrlCode);
end;

function TService1.GetServiceController: TServiceController;
begin
   Result := ServiceController;
end;

procedure TService1.ExecuteService;
begin
   VideoGrabber.CaptureFrameTo (fc_JpegFile, '');
end;

procedure TService1.ServiceCreate(Sender: TObject);
begin
   VideoGrabber := TVideoGrabber.Create (nil);
	VideoGrabber.StoragePath := 'c:\mycapture';
end;

procedure TService1.ServiceDestroy(Sender: TObject);
begin
	VideoGrabber.Free;
end;

procedure TService1.ServiceExecute(Sender: TService);
begin
	while not Terminated do begin
		ExecuteService;
		ServiceThread.ProcessRequests (true);
		Sleep (5000);
	end;
end;

procedure TService1.ServiceStart(Sender: TService; var Started: Boolean);
begin
	VideoGrabber.StartPreview;
end;

procedure TService1.ServiceStop(Sender: TService; var Stopped: Boolean);
begin
	VideoGrabber.StopPreview;
end;

end.

File: CapturarFoto/Componente/vidgrab_8.5.3.1/CBuilder/DEMOS/ServiceApplication/CaptureService_.cpp
//---------------------------------------------------------------------------
#include "CaptureService_.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"

TService1 *Service1;
//---------------------------------------------------------------------------
__fastcall TService1::TService1(TComponent* Owner)
	: TService(Owner)
{
}

//---------------------------------------------------------------------------
TServiceController __fastcall TService1::GetServiceController(void)
{
	return (TServiceController) ServiceController;
}

//---------------------------------------------------------------------------
void __stdcall ServiceController(unsigned CtrlCode)
{
	Service1->Controller(CtrlCode);
}

//---------------------------------------------------------------------------
void __fastcall TService1::ExecuteService (void)
{
   VideoGrabber->CaptureFrameTo (fc_JpegFile, "");
}

//---------------------------------------------------------------------------
void __fastcall TService1::ServiceCreate(TObject *Sender)
{
   VideoGrabber = new TVideoGrabber((TComponent*) NULL);
   VideoGrabber->StoragePath = "c:\mycapture";
}

//---------------------------------------------------------------------------
void __fastcall TService1::ServiceDestroy(TObject *Sender)
{
	delete (VideoGrabber);
}
//---------------------------------------------------------------------------
void __fastcall TService1::ServiceExecute(TService *Sender)
{
	while (! Terminated) {
                ExecuteService();
		ServiceThread->ProcessRequests (true);
		Sleep (5000);
	}
}
//---------------------------------------------------------------------------
void __fastcall TService1::ServiceStart(TService *Sender, bool &Started)
{
	VideoGrabber->StartPreview();
}
//---------------------------------------------------------------------------
void __fastcall TService1::ServiceStop(TService *Sender, bool &Stopped)
{
	VideoGrabber->StopPreview();
}
//---------------------------------------------------------------------------

